httpcore/_async/http_proxy.py

```diff
@@ -39,8 +33,8 @@ class AsyncHTTPProxy(AsyncConnectionPool):
     def __init__(
         self,
         proxy_origin: Origin,
+        proxy_mode: str,
         proxy_headers: Headers = None,
-        proxy_mode: str = "DEFAULT",
```

Comment:
I guess we ought to keep this as `proxy_mode: str = "DEFAULT"` within the scope of this PR.